Examples/test-suite/python/python_overload_simple_cast_runme.py

```diff
@@ -165,8 +165,8 @@ def __float__(self):
 
 
 # unsigned long long
-ullmax = 9223372036854775807  # 0xffffffffffffffff
-ullmaxd = 9007199254740992.0
+ullmax = 0xffffffffffffffff
+ullmaxd = float(2**53 - 1)  # 2**53 does not differ from 2**52+1
```

Comment:
While I ponder this patch...

1. I don't understand the comment; my python interpreter does not agree with it either:
```
>>> 2**53
9007199254740992
>>> 2**52+1
4503599627370497
```
2. Why "_Decrease the range of double->longlong casts by one. max_d+1.0 will now fail._" ? 9007199254740992 is the maximum value that does accurately convert to float, see below, but you've changed that to one less.
```
>>> ll=9007199254740991
>>> float(ll)
9007199254740991.0
>>> ll=9007199254740992
>>> float(ll)
9007199254740992.0
>>> ll=9007199254740993
>>> float(ll)
9007199254740992.0
```

Reply:
Thanks a lot for your review.

1. is a typo. I've corrected it to `2**53 does not differ from 2**53+1`.
2. I actually wonder about the range as well. I think here are two conflicting definitions of max_d. Shall it be the maximum double which can be converted to an integer, or shall `max_d+1` be the first one which cannot be converted to an integer? I think both statements have a valid point, and I'm of course fine with both. Please tell me which way you like more and I will update my patch.

Reply:
From https://en.wikipedia.org/wiki/Double-precision_floating-point_format 

> Between 2<sup>52</sup>=4,503,599,627,370,496 and 2<sup>53</sup>=9,007,199,254,740,992 the representable numbers are exactly the integers. For the next range, from 2<sup>53</sup> to 2<sup>54</sup>, everything is multiplied by 2, so the representable numbers are the even ones, etc. Conversely, for the previous range from 2<sup>51</sup> to 2<sup>52</sup>, the spacing is 0.5, etc. 

So I think `2**53` should convert. `2**53+1` should not. Anything between `2**53` and `2**53+1` should not either as it is not a whole number. Because whole numbers should not convert, I don't see that there is a real difference in the two definitions of `max_d`.

Reply:
That is exactly the point. `2**53+1` does not exist, the closed two numbers are `2**53` and `2**53+2`. Allowing `2**53` will also convert `float(2**53+1) =: 2**53`. So we have to chose between "`2**53+1` should not" and "`2**53` should", both is not possible.

Reply:
Okay, agreed then that `2**53-1` is the the highest whole number that can be uniquely represented as a Python float / IEEE-754 C double and be converted into an integral number.